src/features/diagnosticsProvider.ts

```diff
@@ -319,36 +324,50 @@ class DiagnosticsProvider extends AbstractSupport {
         });
     }
 
-    private static _shouldInclude(quickFix: protocol.QuickFix): boolean {
-        const config = vscode.workspace.getConfiguration('csharp');
```

Comment:
Shouldnt we retain this check here ?

Reply:
I think i removed it because it conflicts with new feature, many of 'unnecessary' diagnostics are 'hidden' by default (they should not show any other thing in UI except faded text, for example should not be visible in problems list etc). If they were supressed -> faded didn't work. 

VScode doesn't seem to have 'hidden' level at all, however if i set level as 'hint' and add fading tag on vscode diagnostic it isn't visible in any way except it's faded. 

Do you know origin of suppressHiddenDiagnostics? Originally it seemed be so that hidden was mapped as information diagnostics (which is bit strange, it is meant to be hidden but then it was elevated as visible) and this flag suppressed all hidden diagnostics.

Reply:
Based on https://github.com/OmniSharp/omnisharp-vscode/issues/1429 `suppressHiddenDiagnostics` is solely to provide some kind of support for unused diagnostics before VSCode implemented faded out code.

I propose that lets remove `suppressHiddenDiagnostics` at this point and add hard coding for unused using. This way end user get better experience (fade out without analyzer support instead of warning) and it doesn't conlflict with version analyzers enabled. Does this shoud good idea @akshita31 ? With ruleset support user can configure any error level for unused usings if they want so (error even), i think this is on same line with visual studio behavior.

Reply:
![image](https://user-images.githubusercontent.com/481896/55970866-0d9a4600-5c89-11e9-8918-3cb6f6c8af93.png)

Ended up following: Instead of promoting hidden as information it now promotes it 'hint'. This can be suppressed `suppressHiddenDiagnostics `. 

This gives meaningfull feature, now its easy to see for example where code actions are available. In same time it's suppressed as default because it causes kind of lot of clutter around -> is optional for those who like it. Suppression was default on original version too.

This way it doesn't conflict with analytics since they really don't have 'hint' level at all, it's just shows them now as small gray dots on screen.